operator/cmd/mesh/uninstall.go

```diff
@@ -103,50 +102,42 @@ func uninstall(cmd *cobra.Command, rootArgs *rootArgs, uiArgs *uninstallArgs, lo
 		return err
 	}
 
-	if err := uninstallWrapper(cmd, rootArgs, uiArgs, restConfig, clt, l); err != nil {
-		return fmt.Errorf("failed to uninstall control plane: %v", err)
-	}
-	return nil
-}
-
-// uninstallWrapper is helper function to uninstalls control plane by either
-// pruning by target revision or deleting specified manifests.
-func uninstallWrapper(cmd *cobra.Command, rootArgs *rootArgs, uiArgs *uninstallArgs,
-	restConfig *rest.Config, clt client.Client, l clog.Logger) error {
 	cache.FlushObjectCaches()
 	opts := &helmreconciler.Options{DryRun: rootArgs.dryRun, Log: l, ProgressLog: progress.NewLog()}
 	h, err := helmreconciler.NewHelmReconciler(clt, restConfig, nil, opts)
 	if err != nil {
 		return fmt.Errorf("failed to create reconciler: %v", err)
 	}
 
-	if uiArgs.revision != "" {
+	// If only revision flag is set, we would prune resources by the revision label.
+	// Otherwise we would merge the revision flag and the filename flag and delete resources by generated manifests.
+	if uiArgs.revision != "" && uiArgs.filename == "" {
```

Comment:
this should be just if uiArgs.filename == "" 
revision must be set if filename is unset, and i'd do these types of syntax checks before you do any other work in the command. the first step should be to validate the params and raise any errors there.